src/mono/mono/mini/interp/transform.c

```diff
@@ -3416,6 +3417,8 @@ interp_emit_swiftcall_struct_lowering (TransformData *td, MonoMethodSignature *c
 	for (int idx_param = 0; idx_param < csignature->param_count; ++idx_param) {
 		MonoType *ptype = csignature->params [idx_param];
 		MonoClass *klass = mono_class_from_mono_type_internal (ptype);
+		MonoGenericClass *gklass = mono_class_try_get_generic_class (klass);
+
 		// SwiftSelf, SwiftError, and SwiftIndirectResult are special cases where we need to preserve the class information for the codegen to handle them correctly.
 		if (mono_type_is_struct (ptype) && !(klass == swift_self || klass == swift_error || klass == swift_indirect_result)) {
 			SwiftPhysicalLowering lowered_swift_struct = mono_marshal_get_swift_physical_lowering (ptype, FALSE);
```

Comment:
Does this lower as `SwiftSelf<T>` instead of as the inner element type? That might be problematic since the layout of `SwiftSelf<T>` may not match the layout of `T` due to potentially added padding at the end.

Reply:
I think you are right, it seems we are lowering the generic, not the inner type. Could you share more details about when padding would be added?

Reply:
Are you referring that it should first retrieve the inner type if `klass` is `SwiftSelf<T>`, and then perform the lowering?

Reply:
> Could you share more details about when padding would be added?

For example, the Swift struct
```swift
struct Foo {
  let A : Int32
  let B : UInt8
}
```
corresponds to the C# struct
```csharp
[StructLayout(LayoutKind.Sequential, Pack = 1)]
struct Foo
{
  public int A;
  public byte B;
}
```

But we have
`Foo` - size 5
`SwiftSelf<Foo>` - size 8

Does this actually matter for the final lowering and ABI? Not totally sure when "self" is the last argument, actually. I wouldn't be surprised if there are cases where lowering `SwiftSelf<Foo>` instead of `Foo` ends up with some different register assignments.

> Are you referring that it should first retrieve the inner type if `klass` is `SwiftSelf<T>`, and then perform the lowering?

Right.